src/KMeans/index.js

```diff
@@ -42,140 +58,156 @@ async function loadDataset(inputData) {
   } else {
     data = inputData;
   }
-  const dataFlat = data.map((d) => {
+  return data.map((d) => {
     return Object.values(d)
   });
-  return dataFlat;
 }
 
-
+/**
+ * // listed on https://learn.ml5js.org/
+ * @property {KMeansOptions} config
+ * @property {Array<[number, number] & {tensor: tf.Tensor<tf.Rank.R1>, centroid: number}>} dataset
+ * @property {tf.Tensor} dataTensor
+ * @property {tf.Tensor2D} centroids
+ *
+ * // not listed
+ * @property {Promise} ready
+ * @property {tf.Tensor2D} centroidsOld
+ */
 class KMeans {
   /**
    * Create a K-Means.
-   * @param {String || array || object} dataset - The dataset to cluster.
-   * @param {options} options - An object describing a model's parameters:
+   * @param {(string | KMeansDataSet)} dataset - The dataset to cluster.
+   * @param {Partial<KMeansOptions>} [options] - An object describing a model's parameters:
    *    - k: number of clusters
    *    - maxIter: Max number of iterations to try before forcing convergence.
-   *    - threshold: Threshold for updated centriod distance before declaring convergence.
-   * @param {function} callback  - Optional. A callback to be called once 
+   *    - threshold: Threshold for updated centroid distance before declaring convergence.
+   * @param {function} [callback] - Optional. A callback to be called once
    *    the model has loaded. If no callback is provided, it will return a 
    *    promise that will be resolved once the model has loaded.
    */
-  constructor(dataset, options, callback) {
+  constructor(dataset, options = {}, callback) {
     this.config = {
-      k: options.k || DEFAULTS.k,
-      maxIter: options.maxIter || DEFAULTS.maxIter,
-      threshold: options.threshold || DEFAULTS.threshold
+      ...options,
+      ...DEFAULTS
     };
     this.ready = callCallback(this.load(dataset), callback);
   }
 
   /**
    * Load dataset, find initial centroids, and run model.
-   * @param {string || array || object} dataset 
+   * Returns a Promise which resolves to the model instance when ready
+   *
+   * @public
+   * @param {(string | KMeansDataSet)} dataset
+   * @return {Promise<this>}
```

Comment:
positive affirmation:
* Thank you SO much for diving into these JSDocs and typing them out 😍